### PR TITLE
Use LiteLLM to add support for Azure, OpenAI, Palm, Anthropic, Cohere Models

### PR DESCRIPTION
**Pull request by ishaan-jaff**
### Background

I'm the maintainer of litellm https://github.com/BerriAI/litellm - a simple & light package to call OpenAI, Azure, Cohere, Anthropic API Endpoints

This PR adds support for models from all the above mentioned providers

Here's a sample of how it's used:

```
from litellm import completion

## set ENV variables
# ENV variables can be set in .env file, too. Example in .env.example
os.environ["OPENAI_API_KEY"] = "openai key"
os.environ["COHERE_API_KEY"] = "cohere key"

messages = [{ "content": "Hello, how are you?","role": "user"}]

# openai call
response = completion(model="gpt-3.5-turbo", messages=messages)

# cohere call
response = completion("command-nightly", messages)

# anthropic call
response = completion(model="claude-instant-1", messages=messages)
```

### PR Quality Checklist
- [x] My pull request is atomic and focuses on a single change.
- [ ] I have thoroughly tested my changes with multiple different prompts.
- [ ] I have considered potential risks and mitigations for my changes.
- [ ] I have documented my changes clearly and comprehensively.
- [ ] I have not snuck in any "extra" small tweaks changes. 
- [ ] I have run the following commands against my code to ensure it passes our linters:
    ```shell
    black .
    isort .
    mypy
    autoflake --remove-all-unused-imports --recursive --ignore-init-module-imports --ignore-pass-after-docstring autogpt tests --in-place
    ```

### CODE REVIEW

**Comment by ishaan-jaff** [comment]
cc @collijk @erik-megarad can I get feedback on this approach?

**Comment by jonasdebeukelaer** [comment]
Couple queries here:
1. how should a user select the llm they want to use? env vars?
2. documentation on how to do the above
3. what will you do with the kwargs that are being passed through

**Comment by ishaan-jaff** [comment]
> how should a user select the llm they want to use? env vars?
users pass in the model using the `model` arg in completion()
eg usage 
```
# openai call
response = completion(model="gpt-3.5-turbo", messages=messages)

# cohere call
response = completion("command-nightly", messages)

# anthropic call
response = completion(model="claude-instant-1", messages=messages)
```

> documentation on how to do the above: 

(Goal of the docs is to be extremely simple)
- litellm github readme: https://github.com/BerriAI/litellm
- litellm read the docs: https://litellm.readthedocs.io/en/latest/

> what will you do with the kwargs that are being passed through
litellm allows users to pass in the same input as chatCompletion.create(). If users are using non OpenAI models, litellm manages the translation of kwargs for them

File: autogpt/llm/providers/openai.py [hunk 1]
```diff
@@ -228,7 +229,7 @@ def create_chat_completion(
         OpenAIObject: The ChatCompletion response from OpenAI
 
     """
-    completion: OpenAIObject = openai.ChatCompletion.create(
```
Comment (ntindle) [inline autogpt/llm/providers/openai.py]: I have concerns that we are directly replacing the provider here without an architecture to support this.

We do a lot more than call the completion such as tracking costs.

Comment (ghost) [reply autogpt/llm/providers/openai.py]: hmm .. given community effort to create similar variants with other providers (e.g. https://github.com/LostAbaddon/AutoClaude) 

i'd be curious to see how we could do that in a 'safe' manner?

Comment (ghost) [reply autogpt/llm/providers/openai.py]: maybe if there's an 'other' provider class - we could potentially use something like 'litellm' to handle the abstraction calling there 

happy to work with @ishaan-jaff on enabling this.

Comment (ntindle) [reply autogpt/llm/providers/openai.py]: I could see that working pretty well. @collijk ?

**Comment by Pwuts** [comment]
I'll look into this soon

**Comment by Pwuts** [comment]
Unfortunately time hasn't been kind to this PR. It would be easier to start over (with the newest version of LiteLLM) than to try to fix up this PR.

At least LiteLLM is doing well. Thanks for taking the time to make this PR @ishaan-jaff, maybe we'll integrate it in the future. :)

Related:
* https://github.com/Significant-Gravitas/AutoGPT/discussions/6969

**Pull request closed without merging**